sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2527,13 +2538,13 @@ def is_provided_ips_valid(ips: Optional[List[Optional[str]]]) -> bool:
                 worker_ip_max_attempts=max_attempts,
                 get_internal_ips=True)
 
-        assert len(cluster_external_ips) == len(cluster_internal_ips), (
+        assert len(cluster_feasible_ips) == len(cluster_internal_ips), (
             f'Cluster {self.cluster_name!r}:'
             f'Expected same number of internal IPs {cluster_internal_ips}'
-            f' and external IPs {cluster_external_ips}.')
+            f' and external IPs {cluster_feasible_ips}.')
```

Comment:
Should this be external -> feasible?

Reply:
This should be `external` as it should not reach here if `use_internal_ips=True`